pkg/build/frontend.go

```diff
@@ -377,3 +394,26 @@ func solvePlatform(ctx context.Context, bopts *BOpts, pl ocispecs.Platform, c ga
 	}
 	return ref, cfgJSON, nil
 }
+
+func globalArgs(buildPlatform, targetPlatform ocispecs.Platform, buildArgs map[string]string, target string) utils.MapGetter {
+	if target == "" {
+		target = "default"
+	}
+	args := map[string]string{
+		"BUILDPLATFORM":   platforms.Format(buildPlatform),
```

Comment:
Why not `FormatAll` here? 

`FormatAll` adds variant, while `Format` omits it

Reply:
These lines are exactly imitating the implementation of BuildKit itself, more specifically https://github.com/moby/buildkit/blob/47c8431ff4e946c0136b22a7cc708702385cff34/frontend/dockerfile/dockerfile2llb/platform.go#L39-L67

I went through the commit history and it looks like `TARGETPLATFORM` got updated when OS version (not variant) is introduced while `BUILDPLATFORM` was kept unmodified. The reason of doing so is unclear, but I tent to preserve the compatibility unless we have a clear motivation.